### ISSUE TITLE
Search: support `path:` parameter

### ISSUE THREAD

**Issue by stsewd** [readthedocs/readthedocs.org#10569]
## What's the problem this feature will solve?

Searching will return results from all files in a set of docs, it could be useful to restrict that search to a given path/directory.

## Describe the solution you'd like

Implement a new `path` parameter https://docs.readthedocs.io/en/stable/server-side-search/syntax.html#parameters, similar to GitHub https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#path-qualifier (well, maybe not that similar, they allow regex :D).

## Alternative solutions

just deal with with a bunch of unrelated results, maybe play with search ranking?

## Additional context

ref https://github.com/readthedocs/readthedocs.org/issues/5966

**Comment by ckreibich** [comment]
Yes please! We'd love to have this in the [Zeek documentation](https://docs.zeek.org/en/master/) to distinguish between the handwritten, book-style documentation and the included, auto-generated documentation based on code docstrings.

**Comment by dracos** [comment]
Just wanted to add that having support for this would be great – I'd love to be able to let people filter/search by different sections of the documentation, if they know they're looking for something in a particular section or want to exclude a section.
